src/RFC-0181_BulletproofsPlus.md

```diff
@@ -0,0 +1,160 @@
+Because the preprint uses the notation $A$ differently in the weighted inner product and range proving protocols, we rename it to $A'$ in the weighted inner product protocol.
+
+Finally, we note one additional unfortunate notation change that applies to the implementation.
+Both the Bulletproofs+ and Zarcanum preprints use $G$ as the commitment value generator, and either $H$ or $\vec{H}\_c$ (in our notation) for masking.
+However, in the Tari protocol (as in other similar protocols), this notation is switched!
+This is because of how generators are defined and used elsewhere in the protocol and elliptic curve library implementation.
+The only resulting change is a switch in generator notation in the Tari implementation: $H$ for the value component generator, and $\vec{G}\_c$ (in our notation) for masking.
+
+## Minimum value assertion
+We first briefly note how to achieve minimum value assertion.
+Let $0 \leq v\_{\text{min}} \leq v \leq v\_{\text{max}}$, where $v\_{\text{min}}$ is a minimum value specified by the prover, $v$ is the value bound to the prover's commitment $V$, and $v\_{\text{max}}$ is a globally-fixed maximum value.
+When generating the proof, the prover uses $v - v\_{\text{min}}$ as the value witness, and additionally binds $v\_{\text{min}}$ into the Fiat-Shamir transcript.
+When verifying the proof, the verifier uses $V - v\_{\text{min}} G_c$ as the commitment (but binds the original commitment $V$ in the transcript).
+This asserts that $v\_{\text{min}} \leq v \leq v\_{\text{min}} + v\_{\text{max}}$.
+While this approach modifies the upper bound allowed for value binding, it does not pose problems in practice, as the intent of range proving is to ensure no overflow when performing balance computations elsewhere in the Tari protocol.
+
+## Extended commitments and aggregation
+
+We now describe how to reduce verification of a single aggregated range proof using extended commitments to a single multiscalar multiplication operation.
+A partial approach is described in the Bulletproofs+ preprint.
+The single multiscalar multiplication used to verify an aggregated range proof can be written more explicitly in our case by accounting for the extra steps used to support extended commitments, and by noting that the $P$ input term to the weighted inner product argument is replaced by the term $\widehat{A}$ defined in the overall range proving protocol.
+
+Suppose we index the inner product generator vectors $\vec{G}$ and $\vec{H}$ using $i$, the inner product recursion generator vectors $\vec{L}$ and $\vec{R}$ using $j$, the aggregated commitment vector $\vec{V}$ by $k$, and the extended commitment mask generator vector $\vec{H}\_c$ by $l$.
+We assume indexing starts at zero unless otherwise noted.
+Single aggregated proof verification reduces to the following:
```

Comment:
Again, a reference to the Zarcanum/BP+ pre-print would help, and everywhere else